src/services/messaging.js

```diff
@@ -37,6 +39,9 @@ function processMessage(req, res) {
     return;
   }
 
+  db.upsert(MESSAGES, messageBundle, r => r.id === messageBundle.id);
+  debug(`Message bundle ${messageBundle.id} added to database`);
+
   forwardMessageResponse(messageBundle).then(() =>
     debug(`Response to /Bundle/${messageBundle.id} forwarded to EHR`)
   );
```

Comment:
It might be a good idea to add logging to `sendOperationOutcome` function below so whenever one is returned we have a log. This could either be a log statement or even storing the OperationOutcome resource in the collection.